wallet/walletrpc.c

```diff
@@ -82,17 +112,49 @@ static u8 *scriptpubkey_from_address(const tal_t *cxt, bool *testnet,
 {
 	struct bitcoin_address p2pkh_destination;
 	struct ripemd160 p2sh_destination;
+	int witness_version;
+	uint8_t witness_program[40];
```

Comment:
Where does the 40 come from? Is this just a buffer that is probably large enough? What happens if we try to decode more?

Reply:
https://github.com/ElementsProject/lightning/blob/master/common/bech32.h#L55 indicates "Pointer to a buffer of size 40"

Looking at the actual `segwit_addr_decode` in detail suggests that the maximum input length is 90 characters, and that the data length cannot be more than 65 characters.  65 characters is 325 bits, but the first character is the witness version.  So that leaves 64 characters for 320 bits, or 40 bytes.

So the documentation of `segwit_addr_decode` seems correct, and the fixed-side buffer is exactly the size indicated in the documentation.

Reply:
I will add a comment to that effect near this area, as well as document the parameters of the `segwit_addr_net_decode` which calls `segwit_addr_decode`, to help future readers/auditors not panic at the fixed-size buffer.

Reply:
Thanks :)    I was about to merge, but if you want do do that I'll wait?

Reply:
Already added at this point, please feel free to merge and thank you.